### PR TITLE
feat: add alignment options to Modal and ScreenCover

### LINKED ISSUE #576: Modal positioning
During the demo of the new data visualizer options it was clear that the height of the modal changing between tabs was not a user friendly experience. I've been thinking about how this can be avoided and what the component itself can do to help minimize this issue.

Current solution:
- Set a max height on the modal. This works, but it requires setting an arbitrary value that may or may not make sense, depending on how good of a guess one can make about the content.

Possible solution:
- I do think we should allow for different positioning of the modal. Allowing a 'top-aligned' modal with a standard offset from the top of the viewport would mean that only the action buttons would move around when the height of the modal changes. This would be a much less jarring experience than a vertically centered modal changing height.

Overly complex, uncessary solution:
- I do not think we should try to do any kind of height detection between tabs. This is overly complex and doesn't belong in the ui-core component.

So, in conclusion, I think we should add the following functionality to the modal component:
- Position: centered, top
- if `position = top`, offset = x (x defaults to 72px)

---

I think this will allow for the flexibility we need. Simpler, single page/tab modal views can continue to use centered vertical alignment. Complex, tabbed modals, or modals that display user generated data that prevents a reasonable max-height being set, can use top alignment.

What do you think @dhis2/team-apps?

### PR DESCRIPTION
**Pull request by Mohammer5**
Fixes #576

Question: Should I outsource "positioningPropType" to `src/common-prop-types.js`?

### CODE REVIEW

File: src/Modal/ModalCard.js [hunk 1]
```diff
@@ -5,23 +5,33 @@ import { resolve } from 'styled-jsx/css'
...
-        height: auto;
-        max-height: 95vh;
         flex-direction: column;
+        height: auto;
         left: 50%;
-        max-width: calc(100vw - ${2 * spacers.dp24}px);
-        max-height: calc(100vh - ${2 * spacers.dp24}px);
+        top: 0;
+        max-height: 95vh;
+        max-height: calc(100vh - ${2 * spacersNum.dp64}px);
```
Comment (edoardo) [inline src/Modal/ModalCard.js]: Duplicate rule?

**Comment by Mohammer5** [comment]
Default has changed to top, other than that, will do @HendrikThePendric

**Comment by Mohammer5** [comment]
@edoardo Fixed the issue and outsourced the prop.
@HendrikThePendric renamed. Could you have a look again please

File: src/Modal.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
 import { createPortal } from 'react-dom'
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
 
-import { sizePropType } from './common-prop-types.js'
-import { ScreenCover } from './ScreenCover.js'
 import { ModalCard } from './Modal/ModalCard.js'
+import { ScreenCover } from './ScreenCover.js'
+import { insideAlignment } from './common-prop-types'
+import { sizePropType } from './common-prop-types.js'
```
Comment (HendrikThePendric) [inline src/Modal.js]: on one line

Comment (HendrikThePendric) [reply src/Modal.js]: Also perhaps keep using the naming convention that's already present, i.e. use the `PropType` suffix

File: src/Modal/ModalCard.js [hunk 3]
```diff
@@ -33,19 +45,17 @@ const cardBoxStyle = resolve`
...
-            {cardBoxStyle.styles}
-        </Card>
-    )
-}
+export const ModalCard = ({ children, small, large, position }) => (
+    <Card
+        className={cx(cardBoxStyle.className, 'modal-card', position, {
+            small,
+            large,
+        })}
```
Comment (HendrikThePendric) [inline src/Modal/ModalCard.js]: Perhaps I am overlooking sth, but I don't see how the classList will ever get a `top` class here...

Comment (Mohammer5) [reply src/Modal/ModalCard.js]: Default prop is missing :+1:

**Comment by Mohammer5** [comment]
@HendrikThePendric Could you have a look at this one again?

**Pull request merged**